### ISSUE TITLE
Apple devices without HA-app are not tracked correctly

### ISSUE THREAD

**Issue by HaraldGithub** [gcobb321/icloud3_v3#145]
I've the issue, that devises without HA-App (iPhone & iWatch of my wife) which were not correct tracked when leaving homezone.
I was with my wife on a trip, my devices (iPhone and coupled watch) were tracked correct, devices of my wife (phone&watch) without HA-app not. Also a manual pushed "locate all devices" doesn't change the location of these devises.
Only after "restart icloud3" over the menu button, the devices without HA-app were located correctly.

**Comment by gcobb321** [comment]
iCloud3 uses triggers from the iOS app for zone enter/exit notifications. When that is not available, the only way iCloud3 will know it has left a zone is when the next location update is done. It will ask Apple for the location and then process the results. Using a short inzone interval (15 minutes) can help. iCloud3 also keeps track of devices nearby. When a device with the iosapp gets an exit trigger, it looks at other devices that are nearby that do not have the iosapp and moves them out of the zone. 

Review the Event Log activity for the devices and see if they are moved out of the zone.

some other people are using a network device tracker to issue an icloud3 action locate service call to relocate the device a few minutes after leaving the network. See the iCloud3 docs Other Features Service Call for more info. 

My wife’s sometimes watch has the same issue.

**Comment by HaraldGithub** [comment]
Hi Gary,

 

thank you very much for your quick response.

I use already a very short inzone interval (15 min).

I’ve tried a manual update of the tracking (“locate this device” or “locate all devices”), but nothing happened/changed.

 

So, at the moment it helps me, when I restart icloud3 via automation, when my iphone with HA-App changes it location,

or when the network device tracker of the non HA-app devices recognized disconnection with my local-/home-network.

 

Thank you again for iCloud3 => You are doing a very good job!

 

Harald

 

Von: Gary Cobb ***@***.***>
Antworten an: gcobb321/icloud3_v3 ***@***.***>
Datum: Montag, 5. Juni 2023 um 14:54
An: gcobb321/icloud3_v3 ***@***.***>
Cc: HaraldGithub ***@***.***>, Author ***@***.***>
Betreff: Re: [gcobb321/icloud3_v3] Apple devices without HA-app are not tracked correctly (Issue #145)

 

iCloud3 uses triggers from the iOS app for zone enter/exit notifications. When that is not available, the only way iCloud3 will know it has left a zone is when the next location update is done. It will ask Apple for the location and then process the results. Using a short inzone interval (15 minutes) can help. iCloud3 also keeps track of devices nearby. When a device with the iosapp gets an exit trigger, it looks at other devices that are nearby that do not have the iosapp and moves them out of the zone.

Review the Event Log activity for the devices and see if they are moved out of the zone.

some other people are using a network device tracker to issue an icloud3 action locate service call to relocate the device a few minutes after leaving the network. See the iCloud3 docs Other Features Service Call for more info.

My wife’s sometimes watch has the same issue.

**Comment by blob810** [comment]
I have a similar problem without the HA-iOS App installed. I installed the integration yesterday. Today in left the first the the zone with the integration installed, but it was not triggered. This is the log at the time I left the home zone:

`9:44:20a	   Located > FamShr-9:29:39a/±9m
9:44:25a	   Location Old #2 > 15 min ago, RetryAt-9:44:40a (15 sec)
9:44:40a	   Location Old #3 > 15 min ago, RetryAt-9:44:55a (15 sec)
9:44:55a	   Location Old #4 > 15 min ago, RetryAt-9:45:55a (1 min)
9:45:55a	   Location Old #5 > 16 min ago, RetryAt-9:46:55a (1 min)
9:46:55a	   Location Old #6 > 17 min ago, RetryAt-9:47:55a (1 min)
9:47:55a	   Location Old #7 > 18 min ago, RetryAt-9:48:55a (1 min)
9:48:55a	   Location Old #8 > 19 min ago, RetryAt-9:53:55a (5 min)
9:53:56a	   Updated > FamShr-9:29:39a/±9m → 9:49:09a/±5m (5 min ago), Moved-432 m
9:53:56a	   Location Old #9 > 5 min ago, RetryAt-9:58:55a (5 min)
9:58:56a	   Updated > FamShr-9:49:09a/±5m → 9:54:00a/±5m (5 min ago), Moved-485 m
9:58:56a	   Location Old #10 > 5 min ago, RetryAt-10:03:55a (5 min)
10:03:56a	   Updated > FamShr-9:54:00a/±5m → 9:58:59a/±5m (5 min ago), Moved-1.26 km
10:03:56a	   Location Old #11 > 5 min ago, RetryAt-10:08:55a (5 min)
10:08:56a	   Updated > FamShr-9:58:59a/±5m → 10:04:01a/±5m (5 min ago), Moved-4.98 km
10:08:56a	   Location Old #12 > 5 min ago, RetryAt-10:38:55a (30 min)
10:38:59a	   Updated > FamShr-10:04:01a/±5m → 10:09:00a/±5m (30 min ago), Moved-7.34 km
10:38:59a	   Location Old #13 > 30 min ago, RetryAt-11:08:55a (30 min)
11:09:00a	   Updated > FamShr-10:09:00a/±5m → 10:39:08a/±4m (30 min ago), Moved-25.0 km
11:09:00a	   Location Old #14 > 30 min ago, RetryAt-11:39:00a (30 min)
11:39:05a	   Updated > FamShr-10:39:08a/±4m → 11:09:04a/±7m (30 min ago), Moved-44.9 km
11:39:05a	   Location Old #15 > 30 min ago, RetryAt-12:09:00p (30 min)
12:09:07p	   Updated > FamShr-11:09:04a/±7m → 11:41:04a/±8m (28 min ago), Moved-2.44 km
12:09:07p	   Location Old #16 > 28 min ago, RetryAt-1:09:05p (1 hr)
1:09:10p	   Updated > FamShr-11:41:04a/±8m → 12:09:17p/±8m (60 min ago), Moved-118 m
1:09:10p	   Location Old #17 > 60 min ago, RetryAt-2:09:05p (1 hr)`

**Comment by gcobb321** [comment]
Review the iCloud3 configuration for each device and make sure the iOS App entity has been selected for each device on the Update Devices screen. Look at iC3 Startup Stages 4 & 5 and verify the FamShr & iOS App devices are listed. Also make sure that the iOS App is selected as a data source on the Data Sources - iCloud and iOS App configuration screen. Location Old entries above usually indicates the phone is in a poor cell area.

**Comment by gcobb321** [comment]
Closed with beta 18. Reopen or post a new issue if you still the problem

**Comment by HaraldGithub** [comment]
Hi Gary,

 

how can I update my iCloud version to iCloud3_v3 Beta18c (6/17/2023)?

I couldn’t find any update link in the icloud-menu and HA also finds no update?

 

Harald

 

Von: Gary Cobb ***@***.***>
Antworten an: gcobb321/icloud3_v3 ***@***.***>
Datum: Montag, 19. Juni 2023 um 09:04
An: gcobb321/icloud3_v3 ***@***.***>
Cc: HaraldGithub ***@***.***>, Author ***@***.***>
Betreff: Re: [gcobb321/icloud3_v3] Apple devices without HA-app are not tracked correctly (Issue #145)

 

Closed with beta 18. Reopen or post a new issue if you still the problem

**Comment by ahamro** [comment]
@HaraldGithub, 
I did that by clicking these buttons:

HACS
Integrations
iCloud3 Device Tracker, Version 3 (HA Integration)
3 dots
Redownload

Cheers,

-Ton,

> Hi Gary, how can I update my iCloud version to iCloud3_v3 Beta18c (6/17/2023)? I couldn’t find any update link in the icloud-menu and HA also finds no update? Harald Von: Gary Cobb ***@***.***> Antworten an: gcobb321/icloud3_v3 ***@***.***> Datum: Montag, 19. Juni 2023 um 09:04 An: gcobb321/icloud3_v3 ***@***.***> Cc: HaraldGithub ***@***.***>, Author ***@***.***> Betreff: Re: [gcobb321/icloud3_v3] Apple devices without HA-app are not tracked correctly (Issue #145) Closed with beta 18. Reopen or post a new issue if you still the problem — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by HaraldGithub** [comment]
Thx, work's for me too!
I would appreciate, if HACS and/or HA would announce such an update automatically.

**Comment by gcobb321** [comment]
So would I. I’m wondering if it’s because it’s a custom repository that is not In the HACS standard repository list. I tried to get v3 added and they would not do it. It is pretty solid now (300+ downloads of b17) and the HACS guys said to update the iCloud3 v2 as a prerelease. I hope to do that once the docs are done and the b18 install numbers get above 200. 

I’m trying to also figure out why the iCloud3 README description does display when you go to HACS, select Integration, select +Explore and Download Repositories, type iCloud3 in the search bar and select iCloud3. Everything is set up properly on my system but it only displays the message “The Developer has not provided any information …”. Can anyone else confirm this?

**Comment by JtwoA** [comment]
This is what I'm seeing (attached). I only see 18c because I re-downloaded the repo. It did error when I attempted to install thru HACS so I just did the ol' manual file drop/replace.

John Alan
[image]

01000011 01110101 01110010 01101001 01101111 01110011 01101001
01110100 01111001 00100000 01101001 01110011 00100000 01110100
01101000 01100101 00100000 01101101 01101111 01110011 01110100
00100000 01110110 01100001 01101100 01110101 01100001 01100010
01101100 01100101 00100000 01100110 01100101 01100001 01110100
01110101 01110010 01100101 00100000 01101111 01100110 00100000
01101001 01101110 01110100 01100101 01101100 01101100 01101001
01100111 01100101 01101110 01110100 00100000 01101100 01101001
01100110 01100101 00101110

------- Original Message -------
